core/dbt/include/global_project/macros/materializations/snapshot/strategies.sql

```diff
@@ -106,7 +107,18 @@
         )
     {%- endset %}
 
-    {% set scd_id_cols = [primary_key] + (check_cols | list) %}
+    {% if target_exists %}
+        {% set row_version -%}
+            (
+             select count(*) from {{ snapshotted_rel }}
```

Comment:
What happens if there are _two_ rows with the same primary key in one snapshot operation? Obviously user error, but do we do anything like error about it, or just give bad results?

Reply:
I think that will result in "undefined behavior" -- the exact outcome depends on:
1) if the specified `check_cols` values are duplicated as well, or if they differ across the duplicated primary key
2) the exact state of the database

Whichever way you slice it, dbt is going to do something undesirable if the specified primary key is duplicated in the snapshot operation though. On pg/redshift, you'll end up with dupes in your snapshot table. On snowflake/bq, you should see a `nondeterministic merge` error.

We could:
1. run a uniqueness check on the staging temp table before loading into the destination table
2. use a `row_number()` (or similar) to dedupe records when building the staging table
  - this would solve some problems but create others!
3. leave it to the user to verify that their snapshot `unique_keys` are unique
  - updating documentation
  - making it possible to specify a `schema.yml` spec for snapshots?